src/ProtectedRoute.tsx

```diff
@@ -5,24 +5,37 @@ import ConditionalWrapper from 'components/utils/ConditionalWrapper';
 import { STATIC_ROUTES } from 'utils/routes';
 import { useUser } from 'store/user';
 import { REDIRECT_URI_KEY } from 'common/constants';
+import { usePersona } from 'store/persona';
 
 type OwnProps = Omit<RouteProps, 'component' | 'render' | 'children'> & {
   layout?: (children: any) => React.ReactElement;
   children: React.ReactNode;
 };
 
+// @TODO: KEYCLOACK remove persona and use userInfo when keycloack is done
 const ProtectedRoute = ({ children, layout, ...routeProps }: OwnProps) => {
-  const { userInfo, error } = useUser();
+  // const { userInfo, error: userError } = useUser();
+  const { personaUserInfo, error: personaError } = usePersona();
   const { keycloak } = useKeycloak();
   const RouteLayout = layout!;
-  const userNeedsToLogin = !userInfo || !keycloak.authenticated;
+  // const userNeedsToLogin = !userInfo || !keycloak.authenticated;
   const currentPath = routeProps.path;
 
-  if (error) {
+  if (personaError) {
     return <Redirect to={STATIC_ROUTES.ERROR} />;
   }
 
-  if (userNeedsToLogin) {
+  if (keycloak.authenticated && (!personaUserInfo || !personaUserInfo?.acceptedTerms)) {
```

Comment:
simplify 
.. && (!personaUserInfo || !personaUserInfo?.acceptedTerms)
to
 .. && !personaUserInfo?.acceptedTerms

Reply:
// to verify! but:
if (personaerror) show error
if (!authenticated) show kc login
if (noprofile) registration
if (profile && no terms and condition) show terms and conditions
if (route === login || registration || acceptedterms) show dashboard
else show route
// edge case: a user with correct profile wants to refuse accepted terms and conditions. i don t think it is possible.